docusaurus/docs/React/customization/typescript.mdx

```diff
@@ -11,20 +11,47 @@ the use of [generics](https://www.typescriptlang.org/docs/handbook/2/generics.ht
 In many cases, TypeScript uses [inference](https://www.typescriptlang.org/docs/handbook/type-inference.html) from provided props to establish a component or
 value's type. It is therefore important that the proper generics be applied to the `stream-chat` client instance during construction. The
 [documentation on `stream-chat` TypeScript](https://github.com/GetStream/stream-chat-js#typescript-v2xx) has detailed examples of how this can be accomplished.
-The Stream Chat client takes seven optional generics that correspond to the seven customizable fields that currently exist in `stream-chat`.
+The Stream Chat client takes a type with seven customizable fields that currently exist in `stream-chat`.
 
 ```tsx
-const client = StreamChat.getInstance<
-  AttachmentType,
-  ChannelType,
-  CommandType,
-  EventType,
-  MessageType,
-  ReactionType,
-  UserType
->('YOUR_API_KEY');
+const client = StreamChat.getInstance<StreamChatGenerics>('YOUR_API_KEY');
 ```
 
+where `StreamChatGenerics` can be defined as a type with the seven generics that correspond to the seven customizable fields as follows:
+
+```tsx
+type StreamChatGenerics = {
+  attachmentType: LocalAttachmentType;
+  channelType: LocalChannelType;
+  commandType: LocalCommandType;
+  eventType: LocalEventType;
+  messageType: LocalMessageType;
+  reactionType: LocalReactionType;
+  userType: LocalUserType;
+};
+```
+
+:::note
+`LocalAttachmentType`, `LocalChannelType` etc. are type definitions for their respective key as per your application types necessities.
+
+Eg:
```

Comment:
I think we can get rid of 'Eg:' here. It's clear the below is an example since it's within a note scope